### ISSUE TITLE
Python bulk upload fails if the resource already exists

### ISSUE THREAD

**Issue by gabrielreid** [waylayio/service-scripts#2]
The Python bulk upload script currently fails if a resource already exists.

Although this tool is only intended to be used as a starter example, it seems likely that users will get confused if, for example, they try to upload a small initial dataset for a resource, and then after that works they try to upload a larger dataset.

We should catch the HTTP 409 that is returned if the resource already exists, and then do a `PATCH` call to update the resource if needed.

**Comment by gillesdemey** [comment]
We do an upsert when sending a `PATCH` request so we can just make that the default operation? :)

**Comment by SanderVanhove** [comment]
If we do this, should we first remove the data that is already present under the resource (maybe with a flag in the params to do this)?

**Comment by gabrielreid** [comment]
> If we do this, should we first remove the data that is already present under the resource (maybe with a flag in the params to do this)?

I don't think that uploading new data (or new metadata) should imply that existing data should be removed -- I would keep those two things separate.

For example, I might have two separate CSV files for a single resource, once that contains measurementA, and another that contains measurementB. I would upload these two files using separate runs of the CSV upload tool, and I definitely don't want anything to be deleted.

I think that it's clearer if all write operations are purely additive, and data deletion is a completely separate operation.

**Comment by SanderVanhove** [comment]
Ok, that would also be a lot clearer toward the user.
